src/components/App/App.tsx

```diff
@@ -102,7 +103,9 @@ const App = ({ featureFromRouter, initialMapView, hpCookie }) => {
         <OsmAuthProvider>
           <StarsProvider>
             <EditDialogProvider /* TODO supply router.query */>
-              <IndexWithProviders />
+              <AddLayerDialogProvider>
```

Comment:
Do you have any plans with this context? I think we can make do without it for this case. 

(I needed a context for EditModal to open it from many places, but your dialog should be perhaps opened only from LayerSwitcher, right?)